src/oidmap.h

```diff
@@ -116,10 +137,8 @@ void git_oidmap_insert(git_oidmap *map, const git_oid *key, void *value, int *rv
 size_t git_oidmap_begin(git_oidmap *map);
 size_t git_oidmap_end(git_oidmap *map);
 
-#define git_oidmap_foreach_value(h, vvar, code) { size_t __i;			\
-	for (__i = git_oidmap_begin(h); __i != git_oidmap_end(h); ++__i) {	\
-		if (!git_oidmap_has_data(h,__i)) continue;			\
-		(vvar) = git_oidmap_value_at(h,__i);				\
+#define git_oidmap_foreach_value(h, vvar, code) { size_t __i = 0;		\
```

Comment:
That's unrelated, but I dislike `code` being an argument… (à la `git_array_foreach`).

Reply:
I totally agree. We could just transform the while-loop into a for loop, initializing `__i` in the loop initializer. But let's do this after this PR got merged, it already has a lot of code churn.